lib/features/support/hooks.rb

```diff
@@ -207,6 +215,47 @@ def output_received_requests(request_type)
   end
 end
 
+# Writes each list of requests to a separate file under, e.g:
+# maze_output/failed/scenario_name/errors.log
+def write_requests(scenario)
+  folder1 = File.join(Dir.pwd, 'maze_output')
+  folder2 = scenario.failed? ? 'failed' : 'passed'
+  folder3 = scenario.name.gsub(' ', '_')
+
+  path = File.join(folder1, folder2, folder3)
+
+  STDOUT.puts "Going to write to #{path}"
+
+  FileUtils.makedirs(path)
+
+  request_types = %w[errors sessions builds logs]
+
+  request_types.each do |request_type|
+    list = Maze::Server.list_for(request_type).all
+    next if list.empty?
+
+    list.each do |request|
+      filename = "#{request_type}.log"
+      filepath = File.join(path, filename)
+
+      File.open(filepath, 'w+') do |file|
```

Comment:
This doesn't change per-request so could be outside of the `list.each` to save opening the same file multiple times

Reply:
Good point, I've also added a heading for each request as well to make it easier to navigate through files.